### PR TITLE
Initial noship delivery of com.ibm.websphere.appserver.javaeeClient-8.0 feature

### PR DESCRIPTION
**Pull request by Channyboy**
Initial noship delivery of `com.ibm.websphere.appserver.javaeeClient-8.0` feature with modification to the FeatureDefinitionUtils to handle this _new_ feature as compatible for the _Client_ process.

The `javaeeClient-8.0` includes the updated and new Java EE8 components relevant for the client-container:

- CDI 2.0
- JPA 2.2
- bVal 2.0
- JavaMail 1.6
- Jsonp 1.1
- Jsonb 1.0

### CODE REVIEW

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
All FAT tests passed.
IM test failures are not related to the contents of commit.

File: dev/com.ibm.websphere.appserver.javaeeClient-8.0/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/bin/
```
Comment (aguibert) [inline dev/com.ibm.websphere.appserver.javaeeClient-8.0/.gitignore]: this file can be removed

File: dev/com.ibm.websphere.appserver.javaeeClient-8.0/com.ibm.websphere.appserver.javaeeClient-8.0.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ com.ibm.websphere.appserver.jpa-2.2, \
+ com.ibm.websphere.appserver.beanValidation-2.0, \
+ com.ibm.websphere.appserver.jaxwsClient-2.2, \
+ com.ibm.websphere.appserver.j2eeManagementClient-1.1, \
+ com.ibm.websphere.appserver.jsonp-1.1, \
+ com.ibm.websphere.appserver.jsonb-1.0, \
+ com.ibm.websphere.appserver.javaMail-1.6, \
+ com.ibm.websphere.appserver.jaxb-2.2, \
+ com.ibm.websphere.appclient.appClient-1.0, \
+ com.ibm.websphere.appserver.jdbc-4.1; ibm.tolerates:="4.2", \
```
Comment (aguibert) [inline dev/com.ibm.websphere.appserver.javaeeClient-8.0/com.ibm.websphere.appserver.javaeeClient-8.0.feature]: here we should include jdbc-4.2, and not tolerate jdbc-4.1

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
#build

**Comment by Channyboy** [comment]
The IM Fat failures do not relate to the javaeeClient-8.0 feature/project and the Liberty Build do not relate to the javaeeClient-8.0 either. The FAT tests added for javaeeClient-8.0 feature in the latest commit have passed in build 'Channyboy-812-20171107-1958'.

**Comment by Channyboy** [comment]
@aguibert  May you please review again.

File: dev/com.ibm.ws.clientcontainer.8.0_fat/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/build/
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/.gitignore]: this file can be removed

File: dev/com.ibm.ws.clientcontainer.8.0_fat/fat/src/com/ibm/ws/clientcontainer/fat/FATSuite.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.custom.junit.runner.AlwaysPassesTest;
+import componenttest.topology.impl.LibertyClient;
+import componenttest.topology.impl.LibertyClientFactory;
+
+@RunWith(Suite.class)
+@SuiteClasses({
+                AlwaysPassesTest.class,
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/fat/src/com/ibm/ws/clientcontainer/fat/FATSuite.java]: AlwaysPasses test should not be needed here

File: dev/com.ibm.ws.clientcontainer.8.0_fat/publish/clients/com.ibm.ws.clientcontainer.fat.ClientContainerClient/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+/apps
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/publish/clients/com.ibm.ws.clientcontainer.fat.ClientContainerClient/.gitignore]: this file can be removed

File: dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/CallbackHandlerNoDefaultConstructor.ear/package.properties [hunk 6]
```diff
@@ -0,0 +1,3 @@
+module.pattern=CallbackHandlerNoDefaultConstructor.jar
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/CallbackHandlerNoDefaultConstructor.ear/package.properties]: all of these package.properties files can be removed

File: dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/CallbackHandlerNoDefaultConstructor.jar/package.properties [hunk 7]
```diff
@@ -0,0 +1,2 @@
+build.classpath=
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/CallbackHandlerNoDefaultConstructor.jar/package.properties]: remove file

File: dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/HelloAppClient.ear/package.properties [hunk 8]
```diff
@@ -0,0 +1,3 @@
+module.pattern=HelloAppClient.jar
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/HelloAppClient.ear/package.properties]: remove file

File: dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/HelloAppClient.jar/package.properties [hunk 9]
```diff
@@ -0,0 +1,2 @@
+build.classpath=
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/HelloAppClient.jar/package.properties]: remove file

File: dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/InAppClientContainer.jar/package.properties [hunk 10]
```diff
@@ -0,0 +1,2 @@
+build.classpath=
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/InAppClientContainer.jar/package.properties]: remove file

File: dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/SystemExitClient.jar/package.properties [hunk 11]
```diff
@@ -0,0 +1,2 @@
+build.classpath=
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/SystemExitClient.jar/package.properties]: remove file

File: dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/SystemExitClientNoDD.ear/package.properties [hunk 12]
```diff
@@ -0,0 +1,3 @@
+module.pattern=SystemExitClientNoDD.jar
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/SystemExitClientNoDD.ear/package.properties]: remove file

File: dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/SystemExitClientNoDD.jar/package.properties [hunk 13]
```diff
@@ -0,0 +1,2 @@
+build.classpath=
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/test-applications/SystemExitClientNoDD.jar/package.properties]: remove file

File: dev/com.ibm.ws.clientcontainer.8.0_fat/fat/src/com/ibm/ws/clientcontainer/fat/AbstractTest.java [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+import org.junit.rules.TestName;
+import org.junit.rules.TestRule;
+
+import com.ibm.websphere.simplicity.ProgramOutput;
+
+import componenttest.custom.junit.runner.OnlyRunInJava7Rule;
+import componenttest.topology.impl.LibertyClient;
+import componenttest.topology.impl.LibertyClientFactory;
+
+public abstract class AbstractTest {
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/fat/src/com/ibm/ws/clientcontainer/fat/AbstractTest.java]: Why is an AbstractTest class needed here?  Would be more straightforward if we could roll this into either `AppClientTest` or the FAT framework

File: dev/com.ibm.ws.clientcontainer.8.0_fat/fat/src/com/ibm/ws/clientcontainer/fat/AppClientTest.java [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+public class AppClientTest extends AbstractTest {
+
+    /*
+     * Basic client launch test.
+     * e.g., "client run com.ibm.ws.clientcontainer.fat.ClientContainerClient"
+     * Check if the test application is printing out "Hello Application Client." to the console.
+     */
+    @Test
+    public void testHelloAppClient() throws Exception {
+        client.copyFileToLibertyClientRoot("clients/HelloAppClient/client.xml");
```
Comment (aguibert) [inline dev/com.ibm.ws.clientcontainer.8.0_fat/fat/src/com/ibm/ws/clientcontainer/fat/AppClientTest.java]: Is it really necessary to copy the client.xml in the beginning of each test case?  This should be done automatically by the FAT framework already.

**Pull request merged**